src/auth/ldap_krb/common_utils.hpp

```diff
@@ -0,0 +1,436 @@
+ * modify it under the terms of the GNU Lesser General Public
+ * License version 2.1, as published by the Free Software
+ * Foundation.  See file COPYING.
+ *
+ */
+
+#ifndef COMMON_UTILS_HPP
+#define COMMON_UTILS_HPP
+
+#include <sys/time.h>
+
+#include <atomic>
+#include <chrono>
+#include <condition_variable>
+#include <functional>
+#include <future>
+#include <mutex>
+#include <queue>
+#include <sstream>
+#include <string>
+#include <thread>
+#include <type_traits>
+#include <vector>
+
+
+namespace common_utils {
+
+  using system_clock  = std::chrono::system_clock;
+
+  /// Common constants used.
+  static auto constexpr ZERO(0);
+  static auto constexpr ONE(1);
+  static auto constexpr VECTOR_GROWTH_FACTOR(2);
+  static auto constexpr FIRST_PRINTABLE_CHAR(20);
+  static const char BLANK(' ');
+  static const char DOT('.');
+  static const char OPEN_OID('{');
+  static const char CLOSE_OID('}');
+  static const char COLON(':');
+  static const char SLASH('/');
+  static const std::string PROTO_SEP("://");
+  static const std::string NEWLINE("\n");
+  static const std::string EMPTY_STR("");
+
+  /// All common messages are defined here
+  //-- TODO: Convert (where possible) std::string to std::string_view.
+  static const std::string
+      MSG_FAILED_DNS_RESOLV("Failed to resolve host/DNS name: ");
+  static const std::string MSG_FAILED_CONNECTION("Failed to connect to: ");
+  static const std::string MSG_FAILED_SOCKET("Failed to create socket: ");
+  static const std::string MSG_ERR_CODE(" Error code: ");
+  static const std::string MSG_ERR_MESSAGE(" Message: ");
+
+
+  const auto is_digit_or_blank = [](const char c) {
+      return (std::isdigit(c) or std::isblank(c));
+  };
+
+  const auto is_empty = [](const std::string& s) {
+      return s.empty();
+  };
+
+  const auto is_open_or_close = [](const char c) {
+      return ((c == OPEN_OID) or (c == CLOSE_OID));
+  };
+
+
+  std::string str_trim(const std::string&);
+  auto from_const_str_to_char(const std::string&);
+  auto from_const_ptr_char_to_string(const char*);
+  auto duplicate_ptr_char(const char*str_to_convert);
+  uint32_t auth_trap_error(const uint32_t);
+  timeval from_chrono_to_timeval(const system_clock::time_point&);
+  system_clock::time_point from_timeval_to_chrono(const timeval& time_val);
+
+
+  template<typename Etp_>
+  using enable_enum_t =
+    std::enable_if_t<std::is_enum<Etp_>::value,
+                            std::underlying_type_t<Etp_>>;
+
+  template<typename Etp_>
+  constexpr inline enable_enum_t<Etp_>
+  enum_underlying_value(Etp_ enm) noexcept
+  {
+    return static_cast<std::underlying_type_t<Etp_>>(enm);
+  }
+
+  template<typename Enm, typename Etp_>
+  constexpr inline typename
+  std::enable_if_t<std::is_enum<Enm>::value &&
+                   std::is_integral<Etp_>::value, Enm>::type
+  convert_to_enum(Etp_ value) noexcept
+  {
+    return static_cast<Enm>(value);
+  }
+
+}   //-- namespace common_utils
+
+
+namespace timer {
```

Comment:
This namespace, or its contents, don't seem to be used anywhere in this PR. Please remove.